### ISSUE TITLE
Postings and comments with same behaviour

### ISSUE THREAD

**Issue by ChartTec** [opensource-socialnetwork/opensource-socialnetwork#924]
In OSSN 4.1 and 4.2 there is still the problem, that {ENTER} sends the comment. That´s a big difference between the Wall-Posting and the Comment-Field. 
To enter a line-feed, I have to send the comments with no line-feeds and then I have to edit the comment. 
In my opinion postings and comments should work the same way. This should be fixed in OSSN 4.3

**Comment by lianglee** [comment]
added in v4.3

**Comment by githubertus** [comment]
There's a slight problem with Firefox and giving focus to comment field:

![firefox-comment-problem](https://cloud.githubusercontent.com/assets/9904364/19892369/cddc539e-a044-11e6-8348-afc5489220f1.png)

**Comment by lianglee** [comment]
I think we need min-height: (height in chrome via dev tools)px;

**Comment by githubertus** [comment]
yep, problem fixed this way, thanks.

**Comment by lianglee** [comment]
@ChartTec , So you want that on Enter you should not send comment,  but instead it should add new line?

**Comment by ChartTec** [comment]
Hello,

yes, that´s it. My customers are mostly between 40 an 60 years old (I am 48) and "my generation" uses [Enter] for an new line and not to send postings :-)
I have several complaints from users, they type a line, press [ENTER], and the edit the comments.
It´s not as important that the comments have newlines [<br> in HTML] but I prefer a button and not [SHIFT] [ENTER]
Thank You for Your great work. As my native language isn´t english, my comments sometimes seems to be unfriendly, but they aren´t! I like OSSN but sometimes I get lost in the scripts. I understand PHP and Javascript, but I am not a learned programmer. So I have some problems to understand code not written by myself.

Greetings,

CVlaus Lampert

**Comment by githubertus** [comment]
I am quite satisfied with the new solution (shift-enter gives newline and enter is sending the comment) as people are used to this from Facebook.
I'm the last one to propagate Facebook's style and functionality for Ossn - but your idea having numerous submit buttons for each comment on the newsfeed page will result in a unquiet overall reading experience - at least as long as we're talking about the Goblue theme.
Thus, my idea would be to create a new theme, first, that comes with comment fields popping up on demand only - and then showing your favored submit button.

**Comment by ChartTec** [comment]
If I would better understand how the script works, I could swap the functions for my site: [Enter] gives newline and [SHIFT][ENTER] or another Key-Kombination, for example [ALT][S], is sending the comment. But for now I haven´t solved my problem.

**Comment by githubertus** [comment]
It's not that complicated :)
Line 11 is catching keypress events from the comment field, stored in e

`11: $('#comment-box-' + $container).keypress(function(e) {`

the "which" method is giving back the value. I have added line 12 to log it in my browser console

`12: console.log(e.which);`

With Firefox, press Strg-Shift-K to open the console (Chrome has a similar thingy)

![consolelog](https://cloud.githubusercontent.com/assets/9904364/20983426/b67e355c-bcbc-11e6-9696-d12c29c71f3a.png)

As the screenshot shows, I have entered "1"[Enter]"a", and the corresponding codes appear in the log now.
We don't have to dig into all that stuff on the next lines - the only interesting part is, that something special should happen, if keycode 13 is arriving because [Enter] has been pressed:

`13: if (e.which == 13) {`

and no shift key has been pressed ...

`14: if (e.shiftKey === false) {`

Now, that you want it just the other way round (= [Enter] plus [Shift] means Submit), simply change the "false" to "true" (as I did it already when doing the screenshot). And that's it.

Be aware to change the logic for OssnEntityComments line 62, too.
